packages/react-native-session-replay/ios/Sources/RCTTextViewRecorder.swift

```diff
@@ -0,0 +1,145 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import UIKit
+@_spi(Internal)
+import DatadogSessionReplay
+import React
+
+internal class RCTTextViewRecorder: SessionReplayNodeRecorder {
+    internal var textObfuscator: (SessionReplayViewTreeRecordingContext) -> SessionReplayTextObfuscating = { context in
+        return context.recorder.privacy.staticTextObfuscator
+    }
+
+    internal var identifier = UUID()
+
+    internal let uiManager: RCTUIManager
+
+    internal init(uiManager: RCTUIManager) {
+        self.uiManager = uiManager
+    }
+
+    internal func extractTextFromSubViews(
+        subviews: [RCTShadowView]?
+    ) -> String? {
+        if let subviews = subviews {
+            return subviews.compactMap { subview in
+                if let sub = subview as? RCTRawTextShadowView {
+                    return sub.text
+                }
+                if let sub = subview as? RCTVirtualTextShadowView {
+                    // We recursively get all subviews for nested Text components
+                    return extractTextFromSubViews(subviews: sub.reactSubviews())
+                }
+                return nil
+            }.joined()
+        }
+        return nil
+    }
+
+    public func semantics(
+        of view: UIView,
+        with attributes: SessionReplayViewAttributes,
+        in context: SessionReplayViewTreeRecordingContext
+    ) -> SessionReplayNodeSemantics? {
+        guard let textView = view as? RCTTextView else {
+            return nil
+        }
+
+        var shadowView: RCTTextShadowView? = nil
+        let tag = textView.reactTag
+
+        RCTGetUIManagerQueue().sync {
+            shadowView = uiManager.shadowView(forReactTag: tag) as? RCTTextShadowView
+        }
+
+        if let shadow = shadowView {
+            // TODO: RUM-2173 check performance is ok
+            let text = extractTextFromSubViews(
+                subviews: shadow.reactSubviews()
```

Comment:
Maybe it's cleaner to just extract the text from this view and change subtreeStrategy to record?

Reply:
I think this is not possible here because of the way RN works.
If you take this example:
```jsx
<Text>this is a <Text>nested text</Text></Text>
```

You would have 1 `RCTTextView` (`UIView`) for all the nested text, with 1 corresponding `RCTTextShadowView` (not a `UIView`) that contains the subtree of nested shadow views like so:

```
RCTTextShadowView
| RCTRawTextShadowView (text: "this is a ")
| RCTVirtualTextShadowView
  | RCTRawTextShadowView (text: "nested text")
```

If we were to change subtreeStrategy to record, we would not find any nested view since we only traverse `UIView` tree, and not this shadow view tree.

Reply:
I believe RN uses an attributedString (which isn't public) on the UIView to achieve the rendering.

Reply:
Maybe all we need is `RCTRawTextShadowViewRecorder` ?

Reply:
I can try to catch it this way, but I'm afraid we won't see it as I believe it is not in the UI View tree that is traversed

Reply:
Indeed it won't ever work:
![image](https://github.com/DataDog/dd-sdk-reactnative/assets/8973379/c70d167f-0c18-49b0-844e-dfd398ec276f)

Reply:
I see. It almost asks for separate set of recorders that would work on those RN views in that case.

I mean - I am happy with the current solution. Wondering if we can do anything better in the future when we scale up number of the views.